src/dist/replication/lib/replica_stub.cpp

```diff
@@ -889,6 +889,41 @@ void replica_stub::on_query_replica_info(const query_replica_info_request &req,
     resp.err = ERR_OK;
 }
 
+void replica_stub::on_query_disk_info(const query_disk_info_request &req,
+                                      /*out*/ query_disk_info_response &resp)
+{
+    ddebug("got query disk info request from (%s)", req.node.to_string());
+
+    for (const auto &dir_node : _fs_manager._dir_nodes) {
+        disk_info disk_info;
+        for (auto primary_iter = dir_node->holding_primary_replicas.begin();
+             primary_iter != dir_node->holding_primary_replicas.end();
+             primary_iter++) {
+            disk_info.holding_primary_replicas[primary_iter->first] =
```

Comment:
`=` should be `+=`, right?

Reply:
No, `dir_node->holding_primary_replicas` is map<app_id, set<gpid>>,`disk_info.holding_primary_replicas` is `map<app_id, gpids_size>`